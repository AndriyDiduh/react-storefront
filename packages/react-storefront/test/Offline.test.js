/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React from 'react'
import { mount } from 'enzyme'
import Offline from '../src/Offline'
import Icon from '@material-ui/icons/Close'

describe('Offline', () => {
  it('renders with no props', () => {
    const component = <Offline />
    expect(mount(component)).toMatchSnapshot()
  })

  it('accepts header, message, and Icon props', () => {
    const component = <Offline header="Header" message="Message" Icon={Icon} />
    expect(mount(component)).toMatchSnapshot()
  })
})
