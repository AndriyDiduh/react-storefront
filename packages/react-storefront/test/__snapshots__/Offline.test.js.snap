// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offline accepts header, message, and Icon props 1`] = `
<WithStyles(Offline) header="Header" message="Message" Icon={[Function: ShouldUpdate]}>
  <Offline header="Header" message="Message" Icon={[Function: ShouldUpdate]} classes={{...}} heading="You\\'re offline">
    <div className="Offline-root-1">
      <pure(CloseIcon) className="Offline-icon-2">
        <CloseIcon className="Offline-icon-2">
          <WithStyles(SvgIcon) className="Offline-icon-2">
            <SvgIcon className="Offline-icon-2" classes={{...}} color="inherit" component="svg" fontSize="default" viewBox="0 0 24 24">
              <svg className="MuiSvgIcon-root-5 Offline-icon-2" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true" role="presentation">
                <path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z" />
                <path fill="none" d="M0 0h24v24H0z" />
              </svg>
            </SvgIcon>
          </WithStyles(SvgIcon)>
        </CloseIcon>
      </pure(CloseIcon)>
      <WithStyles(Typography) variant="h6" component="h1" className="Offline-heading-3">
        <Typography variant="h6" component="h1" className="Offline-heading-3" theme={{...}} classes={{...}} align="inherit" color="default" gutterBottom={false} headlineMapping={{...}} noWrap={false} paragraph={false}>
          <h1 className="MuiTypography-root-14 MuiTypography-h6-31 Offline-heading-3">
            You&#39;re offline
          </h1>
        </Typography>
      </WithStyles(Typography)>
      <WithStyles(Typography) variant="caption" className="Offline-message-4">
        <Typography variant="caption" className="Offline-message-4" theme={{...}} classes={{...}} align="inherit" color="default" gutterBottom={false} headlineMapping={{...}} noWrap={false} paragraph={false}>
          <span className="MuiTypography-root-14 MuiTypography-caption-24 Offline-message-4">
            Message
          </span>
        </Typography>
      </WithStyles(Typography)>
    </div>
  </Offline>
</WithStyles(Offline)>
`;

exports[`Offline renders with no props 1`] = `
<WithStyles(Offline)>
  <Offline classes={{...}} heading="You\\'re offline" message="Please check your internet connection" Icon={[Function: ShouldUpdate]}>
    <div className="Offline-root-1">
      <pure(CloudOffIcon) className="Offline-icon-2">
        <CloudOffIcon className="Offline-icon-2">
          <WithStyles(SvgIcon) className="Offline-icon-2">
            <SvgIcon className="Offline-icon-2" classes={{...}} color="inherit" component="svg" fontSize="default" viewBox="0 0 24 24">
              <svg className="MuiSvgIcon-root-5 Offline-icon-2" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true" role="presentation">
                <path fill="none" d="M0 0h24v24H0z" />
                <path d="M19.35 10.04C18.67 6.59 15.64 4 12 4c-1.48 0-2.85.43-4.01 1.17l1.46 1.46C10.21 6.23 11.08 6 12 6c3.04 0 5.5 2.46 5.5 5.5v.5H19c1.66 0 3 1.34 3 3 0 1.13-.64 2.11-1.56 2.62l1.45 1.45C23.16 18.16 24 16.68 24 15c0-2.64-2.05-4.78-4.65-4.96zM3 5.27l2.75 2.74C2.56 8.15 0 10.77 0 14c0 3.31 2.69 6 6 6h11.73l2 2L21 20.73 4.27 4 3 5.27zM7.73 10l8 8H6c-2.21 0-4-1.79-4-4s1.79-4 4-4h1.73z" />
              </svg>
            </SvgIcon>
          </WithStyles(SvgIcon)>
        </CloudOffIcon>
      </pure(CloudOffIcon)>
      <WithStyles(Typography) variant="h6" component="h1" className="Offline-heading-3">
        <Typography variant="h6" component="h1" className="Offline-heading-3" theme={{...}} classes={{...}} align="inherit" color="default" gutterBottom={false} headlineMapping={{...}} noWrap={false} paragraph={false}>
          <h1 className="MuiTypography-root-14 MuiTypography-h6-31 Offline-heading-3">
            You&#39;re offline
          </h1>
        </Typography>
      </WithStyles(Typography)>
      <WithStyles(Typography) variant="caption" className="Offline-message-4">
        <Typography variant="caption" className="Offline-message-4" theme={{...}} classes={{...}} align="inherit" color="default" gutterBottom={false} headlineMapping={{...}} noWrap={false} paragraph={false}>
          <span className="MuiTypography-root-14 MuiTypography-caption-24 Offline-message-4">
            Please check your internet connection
          </span>
        </Typography>
      </WithStyles(Typography)>
    </div>
  </Offline>
</WithStyles(Offline)>
`;
